plugin/src/sidebar.ts

```diff
@@ -255,6 +257,10 @@ reearth.on("message", ({ action, payload }: PostMessageProps) => {
     }
   } else if (action === "triggerCatalogOpen") {
     reearth.ui.postMessage({ action });
+  } else if (action === "saveExpandedFolders") {
+    expandedFolders = [...payload.expandedFolders];
+  } else if (action === "saveDataset") {
+    dataset = { ...payload.dataset };
```

Comment:
```
let expandedFolders: { id?: string; name?: string }[] = [];
let dataset: DataCatalogItem | undefined = undefined;
```

About expandedFolders:
expandedFolders is initialized with an empty array so it will not be undefined.

About dataset
Dataset's type is `DataCatalogItem | undefined`
Please remember that dataset can be set to undefined for example when resetting the the dataset after clicking the about button. so there is not need to check if it is undefined or not.